From 7cc20e4009d262a7de371313ef1b653d15d70444 Mon Sep 17 00:00:00 2001
From: Andi Kleen <andi@firstfloor.org>
Date: Sat, 1 May 2021 08:22:55 -0700
Subject: [PATCH] objtool: Skip files with existing orc section silently

When rebuilding an LTO tree the build system often decides
to rerun objtool orc on an existing .lto.o file again.
objtool then errors out if the orc section already exists
and fails the build.

Just silently skip regenerating orc if it already exists.

Maybe that would be better fixed in the build system,
but I'm not fully sure how.

Signed-off-by: Andi Kleen <andi@firstfloor.org>
---
 tools/objtool/orc_gen.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/objtool/orc_gen.c b/tools/objtool/orc_gen.c
index 7530a24ea179..46f89481ecf7 100644
--- a/tools/objtool/orc_gen.c
+++ b/tools/objtool/orc_gen.c
@@ -238,8 +238,8 @@ int orc_create(struct objtool_file *file)
 	/* Create .orc_unwind, .orc_unwind_ip and .rela.orc_unwind_ip sections: */
 	sec = find_section_by_name(file->elf, ".orc_unwind");
 	if (sec) {
-		WARN("file already has .orc_unwind section, skipping");
-		return -1;
+		/* Don't regenerate if it already exists */
+		return 0;
 	}
 	orc_sec = elf_create_section(file->elf, ".orc_unwind", 0,
 				     sizeof(struct orc_entry), nr);
